src/main/java/org/Simplycodestudio/CinemaWorld/TicketBookingApp/utils/CinemaUtils.java

```diff
@@ -0,0 +1,138 @@
+package org.Simplycodestudio.CinemaWorld.TicketBookingApp.utils;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+
+public class CinemaUtils {
+
+    private static final Integer MAX_SEATS_IN_ROW = 15;
+    private static final Integer MAX_ROWS_SIZE = 6;
+
+    public static Map<Integer, List<Integer>> getMapWithFreeSeats(Map<Integer, List<Integer>> reservedSeats) {
+        Map<Integer, List<Integer>> resultMap = new HashMap<>();
+        reservedSeats.forEach((row, reservedSeatsInRow) -> {
+            Collections.sort(reservedSeatsInRow);
+            List<Integer> resultOfFreeSlotsInRow = new ArrayList<>();
+            Iterator<Integer> iterator = reservedSeatsInRow.iterator();
```

Comment:
A to nie jest coś czego nie powinniśmy używać... Tak od mniej więcej Javy 5 (10 lat temu)?

Reply:
Mówimy o Liście jako typie generycznym? Używam typu ze względu na to żeby lista nie przyjęła niczego innego jak tylko odgórnie wskazany przeze mnie.

Reply:
Mówię o iteratorze :)

Reply:
Iterator miał za zadanie obsłużenie sytuacji w której kolejny element albo się pojawi i wykonanie na nim konkretnych działań oraz wykonanie konkretnych instrukcji, gdy lista rezerwowanych siedzeń w rzędzie się zakończy. Zapewniało mi to lepsze zabezpieczenie kodu.